kmath-core/src/commonMain/kotlin/kscience/kmath/expressions/DifferentiableExpression.kt

```diff
@@ -0,0 +1,21 @@
+package kscience.kmath.expressions
+
+/**
+ * And object that could be differentiated
+ */
+public interface Differentiable<T> {
+    public fun derivative(orders: Map<Symbol, Int>): T
+}
+
+public interface DifferentiableExpression<T> : Differentiable<Expression<T>>, Expression<T>
+
+public fun <T> DifferentiableExpression<T>.derivative(vararg orders: Pair<Symbol, Int>): Expression<T> =
```

Comment:
Overload with `vararg symbols: Symbol` for order 1 can be added, too.

Reply:
No sense in that. People will use first, maximum second derivatives. There is an extension for the first. The second one could be added any moment.